api/db/populate-attendants.js

```diff
@@ -0,0 +1,85 @@
+const { getRandomSubset, createPrompt } = require('./utils');
+
+const User = require('../models/User');
+const Event = require('../models/Event');
+const Attendant = require('../models/Attendant');
+
+let displayLogs;
+
+const createAttendant = async (attendantParams) => {
+  const attendant = new Attendant({
+    event: attendantParams.event,
+    user: attendantParams.user,
+  });
+  await attendant.save();
+
+  if (displayLogs) {
+    console.log('\n', '\x1b[0m', `New attendant created: ${attendant}`);
+  }
+};
+
+const createAttendants = async (verbose) => {
+  console.log('\n', '\x1b[0m', 'Populating attendants collection...');
+
+  let usersList = await User.distinct('_id');
+  if (usersList.length === 0) {
+    console.log(
+      '\x1b[33m',
+      'The users collection does not exist. Aborted attendants collection population.'
+    );
+    return;
+  }
+
+  let eventsList = await Event.find({
+    startDate: { $gte: new Date() },
+  });
+  if (eventsList.length === 0) {
+    console.log(
+      '\x1b[33m',
+      'The events collection does not exist. Aborted attendants collection population.'
+    );
+    return;
+  }
+
+  displayLogs = verbose;
+  let attendantsList = await Attendant.find();
+  const attendantsArray = [];
+
+  let proceed = false;
+  if (attendantsList.length !== 0) {
+    proceed = await createPrompt(
+      'Some events already contain attendees. Would you like to add attendees to events that already contain attendees?'
+    );
+  }
+
+  for (const event of eventsList) {
+    const attendees = attendantsList
+      .filter((attendee) => attendee.event.toString() == event._id.toString())
+      .map((attendee) => attendee.user.toString());
+
+    const spotsLeft = event.attendantsLimit - attendees.length;
+    // const proceedByLimit =  ?
+    //   event.attendantsLimit >= spotsLeft :
+    //   event.attendantsLimit - 1 >= spotsLeft;
+    const proceedByLimit = event.attendantsLimit - 1 >= spotsLeft;
+    if (proceedByLimit || (!proceed && proceedByLimit)) {
+      continue;
+    }
+
+    const potentialAttendees = usersList.filter(
+      (user) => !attendees.includes(user.toString())
+    );
+
+    const subset = getRandomSubset(potentialAttendees, spotsLeft - 1);
+    subset.unshift(event.createdBy.toString());
+
+    for (const user of subset) {
```

Comment:
```suggestion
    const creatorIncluded = attendees.includes(event.createdBy.toString());

    const newAttendees = getRandomSubset(
      potentialAttendees,
      creatorIncluded ? spotsLeft : spotsLeft - 1
    );

    if (!creatorIncluded) newAttendees.unshift(event.createdBy.toString());

    for (const user of newAttendees) {
```
This is what I meant with my earlier comment.

Reply:
And another issue with this is that we have to make sure that `potentialAttendees` does not include the event creator, as, if it did, they could be included twice on `newAtendees`.